justbuild.it-WebApp/src/test/java/justbuild/it/web/app/service/OfferCreationServiceTest.java

```diff
@@ -0,0 +1,107 @@
+package justbuild.it.web.app.service;
+
+import justbuild.it.web.app.entity.Offer;
+import justbuild.it.web.app.entity.User;
+import justbuild.it.web.app.entity.enums.ServiceCategoryEnum;
+import justbuild.it.web.app.repository.OfferFileRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class OfferCreationServiceTest {
+
+    private OfferFileRepository offerFileRepositoryMock;
+    private OfferCreationService offerCreationService;
+    private Offer offer;
+    private final static String TEST_FILEPATH = "offers.json";
+
+
+    @BeforeEach
+    void SetUpTest() {
+        offerFileRepositoryMock = mock(OfferFileRepository.class);
+        offerCreationService = new OfferCreationService(offerFileRepositoryMock);
+
+        offer = new Offer(1L, ServiceCategoryEnum.CONSTRUCTION, "some content", "Warsaw", new User("", "", "Company Name", "example@example.com", "111 111 111"), LocalDateTime.now());
+    }
+
+    @AfterEach
+    void closeTest() {
+        offerFileRepositoryMock = null;
+        offerCreationService = null;
+        offer = null;
+    }
```

Comment:
ja bym trochę inaczej zrobił ten kawałek kodu z przygotowaniem danych testowych co trochę zoptymalizuje to co się w teście robi. Przetestuj coś takiego:

    private static final String TEST_FILEPATH = "offers.json";

    private OfferFileRepository offerFileRepositoryMock;
    private OfferCreationService offerCreationService;
    private static Offer offer;

    @BeforeAll
    static void setUpTest() {
        offer = new Offer(1L, ServiceCategoryEnum.CONSTRUCTION, "some content", "Warsaw", new User("", "", "Company Name", "example@example.com", "111 111 111"), LocalDateTime.now());
    }

    @AfterAll
    static void closeTest() {
        offer = null;
    }

    @BeforeEach
    void setUp() {
        offerFileRepositoryMock = mock(OfferFileRepository.class);
        offerCreationService = new OfferCreationService(offerFileRepositoryMock);
    }